### PR TITLE
feat: Add pre-release and release workflow for integrations

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
## This PR

- Adds a pre-release and release workflow for Keptn integrations

## Usage Example (pre-)release workflow

```yaml
name: Create Pre-Release
on:
  workflow_dispatch:
jobs:
  test:
    strategy:
      matrix:
        platform: [ubuntu-latest]
    runs-on: ${{ matrix.platform }}
    steps:
      - name: Checkout code
        uses: actions/checkout@v2.3.4
      - name: Run tests
        run: echo "No tests configured yet"

  pre-release:
    needs: test
    name: Pre-Release
    uses: keptn/gh-automation/.github/workflows/pre-release-integration.yml@feature/integration-workflows
    # pre-release and release workflow are interchangeable here!

  release-assets:
    name: Release Assets
    needs: pre-release
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2.3.4
      - name: Add release assets
        env:
          RELEASE_TAG: ${{ needs.pre-release.outputs.RELEASE_TAG }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          gh release upload "$RELEASE_TAG" ./LICENSE

```

### CODE REVIEW

File: .github/workflows/release-integration.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          npx standard-version@^9.3.1 --skip.commit
+
+          TAG=$(git describe --tags --abbrev=0)
+
+          # determine tag name
+          echo "::set-output name=tag-name::$TAG"
+
+          # create a commit with the changelog
+          git add CHANGELOG.md
+          git commit -s -m "chore(release): $TAG"
```
Comment (mowies) [inline .github/workflows/release-integration.yml:61]: I always used `build(release)` for the release commit so that it doesn't show up in the changelog since the `build` annotation is ignored in the changelog.

File: .github/workflows/release-integration.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          echo "⚡️ Pushing changes to remote repository..."
+          git push -f --follow-tags --set-upstream origin patch/release-notes-$RELEASE_TAG
+
+
+      - name: Create GitHub Pull Request for release notes
+        env:
+          RELEASE_TAG: ${{ steps.create-release-package.outputs.tag-name }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh pr create --title "chore: Release notes for $RELEASE_TAG" --body "**This is an automated PR for release notes of $RELEASE_TAG!**"
```
Comment (mowies) [inline .github/workflows/release-integration.yml:86]: Why not use the `peter-evans/create-pull-request@v3` action for doing all this manual stuff? I think it would make this code more readable.

File: .github/workflows/release-integration.yml (hunk 1 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/release-integration.yml]: Fixed

File: .github/workflows/release-integration.yml (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/release-integration.yml:86]: Several reasons:
* I ran into a problem with `peter-evans/create-pull-request@v3` which I couldn't fix easily
* Using the `gh` CLI we are independent from another third party vendor (yes, `gh` classifies as a third party vendor too, but we are already using it to create the release)
* IMO the code is better readable if we refer to the `gh` CLI as often as we can (it creates a repetitive pattern), see screenshot:
image

Comment (mowies) [reply .github/workflows/release-integration.yml:86]: In terms of readability, the GH CLI is not my issue but rather all the bash code before that :)
I feel like with a GH action, the variable names immediately show exactly what branch is used, what repo is used, what commit message is used, etc.

Comment (christian-kreuzberger-dtx) [reply .github/workflows/release-integration.yml:86]: The bash code used in the lines before that line is necessary because `standard-version` does not have an option to sign a commit using `git commit -s`. They only have the option to sign it using `git commit -S`, which uses GPG, which is not available unless we configure a certificate for the gh action user (which is absolutely out of scope for this feature).

Signing the commit is necessary because DCO bot requires it and does not make an exception for the gh-action user (also, this is out of scope for this feature).

Long story short, I would suggest using the code as it is, creating an Issue in `standard-version` regarding the signing issue, and refer to the implementation of this issue in our code.

Comment (christian-kreuzberger-dtx) [reply .github/workflows/release-integration.yml:86]: https://github.com/conventional-changelog/standard-version/issues/854

Comment (mowies) [reply .github/workflows/release-integration.yml:86]: Ok I guess then it's fine

**Review by mowies — APPROVED** [review]
